File: exercism/go/go/reverse-string/reverse_string.go
// Package reverse reverses a string
package reverse

// Reverse returns a new string, a reverse of the input
func Reverse(input string) string {
	if input == "" {
		return ""
	}

	res := make([]rune, len(input))
	i := 0
	// store input string in a slice of runes
	for _, r := range input {
		res[i] = r
		i++
	}
	res = res[0:i]

	// reverse rune now
	for j := 0; j < i/2; j++ {
		res[j], res[i-1-j] = res[i-1-j], res[j]
	}

	return string(res)
}

File: exercism/cpp/cpp/reverse-string/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string {
	std::string reverse_string(std::string &input) {
		int len = input.length();
		for (int i = 0; i < len/2; i++) {
			char c = input[i];
			input[i] = input[len-i-1];
			input[len-i-1] = c;
		}
		return input;
	}
}  // namespace reverse_string
